integration-tests/jackson-protobuf/src/main/java/org/apache/camel/quarkus/component/jackson/protobuf/it/UppercaseTextDeserializer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.jackson.protobuf.it;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+/**
+ * Simple deserializer to uppercase the text field on a {@link Pojo} instance
+ */
+public class UppercaseTextDeserializer extends StdDeserializer<Pojo> {
+
+    public UppercaseTextDeserializer() {
+        this(null);
+    }
+
+    public UppercaseTextDeserializer(Class<?> src) {
+        super(src);
+    }
+
+    @Override
+    public Pojo deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+        JsonNode jsonNode = p.getCodec().readTree(p);
+        String text = jsonNode.get("text").asText();
+        Pojo pojo = new Pojo();
+        pojo.setText(text.toUpperCase(Locale.US));
```

Comment:
A detail that will hardly ever matter unless we pass some unusual characters to Pojo.text: we might want to make sure that the locale is available in native mode. I do not think we really need to care in this particular case, but it is good to be aware at least https://github.com/quarkusio/quarkus/issues/5244 https://github.com/apache/camel-quarkus/issues/1861#issuecomment-832824861